### PR TITLE
Reimplement `unmunge` with generator function

### PR DESCRIPTION
**Pull request by tothtamas28**
Improves `kore-to-kast` performance on the profiled instance by 30%.

### CODE REVIEW

**Comment by ehildenb** [comment]
@tothtamas28 can we have it show _all_ durations here? 
image

When there are modifications to test-suites, I like to double-check that the actual test ran on CI.

**Comment by ehildenb** [comment]
I guess the remaining durations are setup/teardown, but still, no reason to hide information on CI. More output makes it easier to diagnose issues before having to checkout the branch locally.

**Review by ehildenb — APPROVED** [review]

File: src/tests/profiling/profile_kast_to_kore.py [hunk 1]
```diff
@@ -22,15 +22,18 @@ def test_kast_to_kore(profile: Profiler) -> None:
...
     kast = KInner.from_json(kinner_file.read_text())
 
-    for file_name in ['first.txt', 'second.txt']:
+    for file_name in ['kast-to-kore-1.txt', 'kast-to-kore-2.txt']:
         with profile(file_name, sort_keys=('cumtime',), limit=25):
-            kast_to_kore(kast_defn, kore_defn, kast, sort=KSort('GeneratedTopCell'))
+            kore = kast_to_kore(kast_defn, kore_defn, kast, sort=KSort('GeneratedTopCell'))
+
+    with profile('kore-to-kast.txt', sort_keys=('cumtime',), limit=25):
+        kore_to_kast(kast_defn, kore)
```
[the remark this answers is not in the document]
Comment (tothtamas28) [reply src/tests/profiling/profile_kast_to_kore.py:39]: I changed them for consistency. Before, this test only profiled KAST-to-KORE, so  calling the output`first.txt` and `second.txt` was fine.

**Comment by tothtamas28** [comment]
> can we have it show all durations here?
> I like to double-check that the actual test ran on CI

I can set it more verbose if you want, but the list of executed tests is already printed.

**Comment by tothtamas28** [comment]
> I guess the remaining durations are setup/teardown, but still, no reason to hide information on CI. More output makes it easier to diagnose issues before having to checkout the branch locally.

Created an issue: https://github.com/runtimeverification/pyk/issues/360

**Pull request merged**